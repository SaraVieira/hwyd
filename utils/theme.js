import {
  /* colors */
  APPLE_VALLEY,
  BREWED_MUSTARD,
  CREAMY_PEACH,
  OLD_GERANIUM,
  PENCIL_LEAD,
  ROGUE_PINK,
  ROSY_HIGHLIGHT,
  SOFT_BLUE,
  SQUEAKY,
  WHITE,

  /* fonts */
  BOLD_FONT,
  ITALIC_FONT,
  REGULAR_FONT,
  SUBTITLE_FONT,
  TITLE_FONT,
} from './constants';

export default {
  colors: {
    appleValley: APPLE_VALLEY,
    brewedMustard: BREWED_MUSTARD,
    creamyPeach: CREAMY_PEACH,
    oldGeranium: OLD_GERANIUM,
    pencilLead: PENCIL_LEAD,
    roguePink: ROGUE_PINK,
    rosyHighlight: ROSY_HIGHLIGHT,
    softBlue: SOFT_BLUE,
    squeaky: SQUEAKY,
    white: WHITE,
  },
  fonts: {
    bold: BOLD_FONT,
    italic: ITALIC_FONT,
    regular: REGULAR_FONT,
    subtitle: SUBTITLE_FONT,
    title: TITLE_FONT,
  },
};
