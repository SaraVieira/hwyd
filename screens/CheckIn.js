/* @flow */

import React, { Component } from 'react';
import { Text, View } from 'react-native';
import styled from 'styled-components';
import { LinearGradient, Constants } from 'expo';
import { Entypo } from '@expo/vector-icons';
import parse from 'date-fns/parse';
import format from 'date-fns/format';
import type {
  NavigationScreenProp,
  NavigationStateRoute,
} from 'react-navigation';

import { Button, ButtonText, Done, Loader } from '../components';
import { getCheckIns, addCheckIn } from '../utils/actions';
import { ROGUE_PINK, ROSY_HIGHLIGHT } from '../utils/constants';
import type { CheckInData } from '../utils/types';

type Props = {
  navigation: NavigationScreenProp<NavigationStateRoute>,
};

type State = {
  isToday: boolean,
  loading: boolean,
  hasCheckInForToday: boolean,
  date: string,
  displayDate: ?string,
};

const ButtonsContainer = styled(View)`
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const Container = styled(View)`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

const ParagraphText = styled(Text)`
  font-family: ${props => props.theme.fonts.regular};
`;

const Title = styled(Text)`
  font-family: ${props => `${props.theme.fonts.title}`};
  font-size: 40;
  margin-bottom: 36;
  text-align: center;
`;

const DisplayDate = styled(Text)`
  font-family: ${props => `${props.theme.fonts.subtitle}`};
  font-size: 24;
  margin-bottom: 16;
  text-align: center;
`;

const Background = styled(LinearGradient)`
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
`;

const BackButton = styled(Button.Transparent)`
  position: absolute;
  top: 32;
  left: -8;
`;

export default class CheckIn extends Component<Props, State> {
  state = {
    isToday: !this.props.navigation.state.params,
    loading: true,
    hasCheckInForToday: false,
    date: this.props.navigation.state.params
      ? this.props.navigation.state.params.date.dateString
      : format(new Date(), 'YYYY-MM-DD'),
    displayDate: this.props.navigation.state.params
      ? format(
          parse(this.props.navigation.state.params.date.dateString),
          'MMMM Do, YYYY',
        )
      : undefined,
  };

  componentWillMount = async () => {
    const checkIns = await getCheckIns();
    const todaysCheckIn = checkIns.find(c => c.date === this.state.date);

    this.setState(() => ({
      loading: false,
      hasCheckInForToday: !!todaysCheckIn,
    }));
  };

  render() {
    return this.state.loading ? (
      <Container>
        <Loader style={{ width: 100, height: 100 }} />
      </Container>
    ) : this.state.hasCheckInForToday ? (
      <Container>
        <Background colors={[ROGUE_PINK, ROSY_HIGHLIGHT]} />

        {!this.state.isToday && (
          <BackButton onPress={() => this.props.navigation.pop()}>
            <Entypo size={24} name="chevron-thin-left" />
          </BackButton>
        )}

        {!this.state.isToday && (
          <DisplayDate>{this.state.displayDate}</DisplayDate>
        )}

        <Title>Another Day Recorded!</Title>

        <Done style={{ width: 200, height: 200 }} />
      </Container>
    ) : (
      <Container>
        <Background colors={[ROGUE_PINK, ROSY_HIGHLIGHT]} />

        {!this.state.isToday && (
          <BackButton onPress={() => this.props.navigation.pop()}>
            <Entypo size={24} name="chevron-thin-left" />
          </BackButton>
        )}

        {!this.state.isToday && (
          <DisplayDate>{this.state.displayDate}</DisplayDate>
        )}

        <Title>How was your day?</Title>

        <ButtonsContainer>
          <Button.Transparent
            style={{ marginRight: 8 }}
            onPress={async () => {
              await addCheckIn({
                date: this.state.date,
                result: 'GOOD',
              });
              this.setState(() => ({ hasCheckInForToday: true }));
            }}
          >
            <ButtonText.Transparent>
              <Entypo size={98} name="emoji-happy" />
            </ButtonText.Transparent>
          </Button.Transparent>

          <Button.Transparent
            onPress={async () => {
              await addCheckIn({
                date: this.state.date,
                result: 'BAD',
              });
              this.setState(() => ({ hasCheckInForToday: true }));
            }}
          >
            <ButtonText.Transparent>
              <Entypo size={98} name="emoji-sad" />
            </ButtonText.Transparent>
          </Button.Transparent>
        </ButtonsContainer>
      </Container>
    );
  }
}
