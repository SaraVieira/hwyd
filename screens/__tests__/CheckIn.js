import React from 'react';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import CheckIn from '../CheckIn';
import theme from '../../utils/theme';

describe('<CheckIn />', () => {
  it('renders loading state', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <CheckIn navigation={{ state: {} }} />
      </ThemeProvider>,
    );

    const root = tree.find(CheckIn).shallow();
    expect(root).toMatchSnapshot();
  });

  it('renders check in state', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <CheckIn navigation={{ state: {} }} />
      </ThemeProvider>,
    );

    const root = tree.find(CheckIn).shallow();
    root.setState({ loading: false, hasCheckInForToday: false });
    expect(root).toMatchSnapshot();
  });

  it('renders done state', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <CheckIn navigation={{ state: {} }} />
      </ThemeProvider>,
    );

    const root = tree.find(CheckIn).shallow();
    root.setState({ loading: false, hasCheckInForToday: true });
    expect(root).toMatchSnapshot();
  });
});
