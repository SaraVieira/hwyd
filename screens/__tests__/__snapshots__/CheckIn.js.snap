// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckIn /> renders check in state 1`] = `
<Styled(View)>
  <Styled(LinearGradient)
    colors={
      Array [
        "#f8a5c2",
        "#f7d794",
      ]
    }
  />
  <Styled(Text)>
    How was your day?
  </Styled(Text)>
  <Styled(View)>
    <Styled(TouchableOpacity)
      onPress={[Function]}
      style={
        Object {
          "marginRight": 8,
        }
      }
    >
      <Styled(Text)>
        <Icon
          allowFontScaling={false}
          name="emoji-happy"
          size={98}
        />
      </Styled(Text)>
    </Styled(TouchableOpacity)>
    <Styled(TouchableOpacity)
      onPress={[Function]}
    >
      <Styled(Text)>
        <Icon
          allowFontScaling={false}
          name="emoji-sad"
          size={98}
        />
      </Styled(Text)>
    </Styled(TouchableOpacity)>
  </Styled(View)>
</Styled(View)>
`;

exports[`<CheckIn /> renders done state 1`] = `
<Styled(View)>
  <Styled(LinearGradient)
    colors={
      Array [
        "#f8a5c2",
        "#f7d794",
      ]
    }
  />
  <Styled(Text)>
    Another Day Recorded!
  </Styled(Text)>
  <Done
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
</Styled(View)>
`;

exports[`<CheckIn /> renders loading state 1`] = `
<Styled(View)>
  <Loader
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</Styled(View)>
`;
